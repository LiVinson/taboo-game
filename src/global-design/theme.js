const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
