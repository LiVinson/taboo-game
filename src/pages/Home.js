import React from "react"
import Modal from "../components/Modal"
import { Redirect } from "react-router-dom"
import JoinForm from "../components/JoinForm"
import { createNewCode } from "../utils/helpers"
import { createNewGame } from "../utils/API"

export default class Home extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      showInstructions: false,
      gamecode: null,
      redirect: false,
    }

    this.style = {
      width: "400px",
      height: "200px",
      fontSize: "50px",
    }
    this.toggleInstructions = this.toggleInstructions.bind(this)
    this.createGameCode = this.createGameCode.bind(this)
  }

  toggleInstructions() {
    this.setState((state) => {
      return {
        ...state,
        showInstructions: !state.showInstructions,
      }
    })
  }

  createGameCode(e) {
    e.target.disabled = true
    const gamecode = createNewCode()
    createNewGame(gamecode)
      .then((res) => {
        //Redirect to waiting room.
        this.setState((state) => ({
          ...state,
          redirect: true,
          gamecode,
        }))
      })
      .catch((err) => {
        console.log(
          "There was an error creating a new game with gamecode: ",
          gamecode
        )
      })
  }

  render() {
    const { redirect, gamecode } = this.state
    if (gamecode && redirect) {
      return (
        <Redirect
          push //Keeps home location on the history stack
          to={{ pathname: "new", search: `?gamecode=${gamecode}` }}
        />
      )
    }

    return (
      <div
        style={{
          display: "flex",
          justifyContent: "space-around",
          paddingTop: "20px",
        }}
      >
        <button onClick={(e) => this.createGameCode(e)} style={this.style}>
          Create New Game
        </button>

        <button
          onClick={() => {
            this.toggleInstructions()
          }}
          style={this.style}
        >
          Join Game
        </button>
        {this.state.showInstructions ? (
          <Modal
            display={this.state.showInstructions}
            toggleDisplay={this.toggleInstructions}
            header="Join a Game of Taboo"
            buttonText=""
          >
            {<JoinForm confirmGame={(code) => {}} />}
          </Modal>
        ) : null}
      </div>
    )
  }
}
