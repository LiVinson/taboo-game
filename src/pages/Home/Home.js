import React from 'react'
import PropTypes from "prop-types"
import { Switch, Route } from 'react-router-dom'
import MainMenu from 'components/MainMenu'
import CreateGame from 'components/CreateGame'
import JoinGame from 'components/JoinGame'
import Rules from 'components/Rules'
import Submit from 'components/Submit'

//return one of series of routes based on current home path
export default function Home({ match }) {
	const { path } = match
	return (
			<Switch>
				<Route path={path} exact component={MainMenu} />
				<Route path={`${path}/create`} component={CreateGame} />
				<Route path={`${path}/join`} component={JoinGame} />
				<Route path={`${path}/rules`} component={Rules} />
				<Route path={`${path}/submit`} component={Submit} />
			</Switch>
	)
}

Home.propTypes = {
	match : PropTypes.object.isRequired
}
