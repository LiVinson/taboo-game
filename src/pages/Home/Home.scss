
// body,
// #root
//  {
//      position:relative;
//     overflow:hidden; 
// }
.wrapper {
    background-image: linear-gradient(to bottom, var(--color-white) 0%, var(--color-white) 50%,
         70%, var(--color-primary) 90%);
    padding: 2rem 2.5rem;
    min-height:100vh;
    position: relative;
    overflow:hidden; 

}

.header {
    background: none;
    font-family: 'Oswald', sans-serif;
    font-size: 3rem;
    color: var(--color-primary);
    text-shadow: 0px 1rem 1rem rgba(0,0,0,.6);
    text-transform:uppercase;
}

//For main homescreen only, display larger text.
.header--large {
    font-size: 6rem;
    text-align: center;
    margin: 1rem 0;
}



.subheading {
    font-size: 2.2rem;
    text-align: center;
    padding: 0 1rem;
    color: var( --color-gray-dark-1);
    margin-bottom: 7rem;
    font-weight: 400;
    font-family: 'Oswald', sans-serif;

}

.subheading__focus-text {
    color: var(--color-primary-light);
}

/* Card Component */

.taboo-card {
    position: relative;
    background-color: var(--color-primary);
    min-width:30rem; 
    padding: 2rem;
    border-radius: 2px;
    box-shadow: 0px .8rem 1rem rgba(0,0,0,.9);
    //Used for inheritance on child components
    color: var(--color-primary-light);
    z-index: 15;


    //Main taboo card on home screen, make text and padding larger
    &--home &__word-container &__word {
        font-size: 4rem;
    }

    &--home &__list-container {
        padding-top:2rem;
        padding-bottom:2rem;
    }

    &--bottom {    
        color:  var( --color-tertiary);
        background-color: currentColor;
        position: absolute;
        bottom:13rem;
        right: 8rem;
        z-index: 5;
        transform: rotate(-45deg);
    }

    &--middle {
        color:  var( --color-secondary);
        background-color: currentColor;
        position: absolute;
        bottom:12rem;
        left:7rem;
        z-index: 10;
        transform: rotate(45deg)
    }

    &__word-container
    {
        background-color: var(--color-off-white);
        padding: 1.2rem;
        text-align: center;   
        margin-bottom: 1.5rem;
        border-radius: 2px;
        color:inherit;

    }

    &__word {
        font-family: 'Oswald', sans-serif;
        font-size: 3.5rem;
        color: inherit;
        font-weight: 600;
        text-transform: uppercase;
    
    }

    &__list-container {
        background-color: var(--color-off-white);
        padding: 1.5rem;
        text-align: center;  
        border-radius: 2px;
    }

    &__list {
        list-style-type: none;

    }

    &__list-item {
        font-family: 'Oswald', sans-serif;
        font-size: 2.5rem;
        position: relative;
        padding-bottom:1.5rem;
    }


    &__list-item:after {
        content:"";
        display:block;
        width: 10rem;
        margin: 0 auto;
        padding-top:1rem;
        border-bottom: #C4C4C4 2px solid; 
    }
}



