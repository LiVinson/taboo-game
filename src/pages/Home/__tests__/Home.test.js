import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Home from '../Home'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'

describe('Home component renders correctly on each subroute', () => {
	const defaultProps = {
		match: {
			path: '/home',
		},
	}

	test('Home renders correctly on /home route', () => {
		const props = {
			...defaultProps
		}
		/* component contains switch, so MR required*/
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<MemoryRouter>
						<Home {...props} />
					</MemoryRouter>
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('Home renders correctly on /home/create route', () => {
		const props = {
			...defaultProps,
			match: {
				props: "/home/create"
			}
		}
		/* component contains switch, so MR required*/
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<MemoryRouter>
						<Home {...props} />
					</MemoryRouter>
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('Home renders correctly on /home/join route', () => {
		const props = {
			...defaultProps,
			match: {
				props: "/home/join"
			}
		}
		/* component contains switch, so MR required*/
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<MemoryRouter>
						<Home {...props} />
					</MemoryRouter>
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('Home renders correctly on /home/rules route', () => {
		const props = {
			...defaultProps,
			match: {
				props: "/home/rules"
			}
		}
		/* component contains switch, so MR required*/
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<MemoryRouter>
						<Home {...props} />
					</MemoryRouter>
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})
})
