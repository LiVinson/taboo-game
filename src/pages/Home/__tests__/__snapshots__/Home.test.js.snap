// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly on each subroute Home renders correctly on /home route 1`] = `null`;

exports[`Home component renders correctly on each subroute Home renders correctly on /home/create route 1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c4 {
  font-family: 'Oswald',sans-serif;
  font-size: 2.7rem;
  position: relative;
  padding-bottom: 1.5rem;
}

.c3 {
  list-style-type: none;
}

.c5 {
  -webkit-transition: all .2s ease-in;
  transition: all .2s ease-in;
}

.c5:link,
.c5:visited {
  color: currentColor;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c5:focus {
  text-shadow: 0px 0.2rem 0.3rem rgba(0,0,0,0.7);
  -webkit-transform: translateY(-3px);
  -ms-transform: translateY(-3px);
  transform: translateY(-3px);
}

.c5:hover {
  -webkit-filter: brightness(1.2);
  filter: brightness(1.2);
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Menu
  </div>
  <div
    className="c2"
  >
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        <a
          className="c5"
          href="//create"
          onClick={[Function]}
          text="Create New Game"
        >
          Create New Game
        </a>
      </li>
      <li
        className="c4"
      >
        <a
          className="c5"
          href="//join"
          onClick={[Function]}
          text="Join Game"
        >
          Join Game
        </a>
      </li>
      <li
        className="c4"
      >
        <a
          className="c5"
          href="//rules"
          onClick={[Function]}
          text="How to Play"
        >
          How to Play
        </a>
      </li>
      <li
        className="c4"
      >
        <a
          className="c5"
          href="//submit"
          onClick={[Function]}
          text="Submit a Card"
        >
          Submit a Card
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Home component renders correctly on each subroute Home renders correctly on /home/join route 1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c4 {
  font-family: 'Oswald',sans-serif;
  font-size: 2.7rem;
  position: relative;
  padding-bottom: 1.5rem;
}

.c3 {
  list-style-type: none;
}

.c5 {
  -webkit-transition: all .2s ease-in;
  transition: all .2s ease-in;
}

.c5:link,
.c5:visited {
  color: currentColor;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c5:focus {
  text-shadow: 0px 0.2rem 0.3rem rgba(0,0,0,0.7);
  -webkit-transform: translateY(-3px);
  -ms-transform: translateY(-3px);
  transform: translateY(-3px);
}

.c5:hover {
  -webkit-filter: brightness(1.2);
  filter: brightness(1.2);
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Menu
  </div>
  <div
    className="c2"
  >
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        <a
          className="c5"
          href="//create"
          onClick={[Function]}
          text="Create New Game"
        >
          Create New Game
        </a>
      </li>
      <li
        className="c4"
      >
        <a
          className="c5"
          href="//join"
          onClick={[Function]}
          text="Join Game"
        >
          Join Game
        </a>
      </li>
      <li
        className="c4"
      >
        <a
          className="c5"
          href="//rules"
          onClick={[Function]}
          text="How to Play"
        >
          How to Play
        </a>
      </li>
      <li
        className="c4"
      >
        <a
          className="c5"
          href="//submit"
          onClick={[Function]}
          text="Submit a Card"
        >
          Submit a Card
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Home component renders correctly on each subroute Home renders correctly on /home/rules route 1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c4 {
  font-family: 'Oswald',sans-serif;
  font-size: 2.7rem;
  position: relative;
  padding-bottom: 1.5rem;
}

.c3 {
  list-style-type: none;
}

.c5 {
  -webkit-transition: all .2s ease-in;
  transition: all .2s ease-in;
}

.c5:link,
.c5:visited {
  color: currentColor;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c5:focus {
  text-shadow: 0px 0.2rem 0.3rem rgba(0,0,0,0.7);
  -webkit-transform: translateY(-3px);
  -ms-transform: translateY(-3px);
  transform: translateY(-3px);
}

.c5:hover {
  -webkit-filter: brightness(1.2);
  filter: brightness(1.2);
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Menu
  </div>
  <div
    className="c2"
  >
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        <a
          className="c5"
          href="//create"
          onClick={[Function]}
          text="Create New Game"
        >
          Create New Game
        </a>
      </li>
      <li
        className="c4"
      >
        <a
          className="c5"
          href="//join"
          onClick={[Function]}
          text="Join Game"
        >
          Join Game
        </a>
      </li>
      <li
        className="c4"
      >
        <a
          className="c5"
          href="//rules"
          onClick={[Function]}
          text="How to Play"
        >
          How to Play
        </a>
      </li>
      <li
        className="c4"
      >
        <a
          className="c5"
          href="//submit"
          onClick={[Function]}
          text="Submit a Card"
        >
          Submit a Card
        </a>
      </li>
    </ul>
  </div>
</div>
`;
